src/security/cert_generators/file/certificate_db.cc

```diff
@@ -283,7 +281,7 @@ bool Ssl::CertificateDb::purgeCert(std::string const & key) {
 bool
 Ssl::CertificateDb::addCertAndPrivateKey(std::string const &useKey, const Security::CertPointer &cert, const Security::PrivateKeyPointer &pkey, const Security::CertPointer &orig)
 {
-    const Locker locker(dbLock, Here);
+    const Locker locker(dbLock, __FILE__, __LINE__);
```

Comment:
* If that filename/lineno information is never used, please remove the corresponding parameters and Locker members.
* Otherwise, please use Here() from `base/Here.h` instead, adding a SourceLocation member to Locker.

I suspect the former is true.

Reply:
Sometimes it is used, sometimes not. It depends on debug settings.

The "Here" is not being altered, it was a #define expanding to "_FILE_, __LINE_" which now has a symbol clash with the base/Here.h Here() definition. That is the only reason this macro is being touched at all. Changes to how the ssl_crtd database operates is out of scope for this PR. So not doing any more than expanding the macro.

Reply:
> Sometimes it is used, sometimes not. It depends on debug settings.

Please give an example where it is used (with any debug settings you need to make that example work).

Reply:
To be precise the values are stored so they can be displayed in stack traces and state traces when debugging a core file when ssl_crtd / security_file_certgen crashes. The debug settings are the OS core enable/disable.

Reply:
I see. Please at least add an XXX then, so that this code is not abused as an excuse to add more ugly `__FILE__/__LINE__` code elsewhere, with even weaker excuses. For example:

```suggestion
    const Locker locker(dbLock, __FILE__, __LINE__); // XXX: Use Here().
```